File: Hospital Management System/frontend/src/services/prescriptionService.js
import api from './api';

const prescriptionService = {
  getAllPrescriptions: async () => {
    try {
      const response = await api.get('/prescriptions');
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  addPrescription: async (prescription) => {
    try {
      const response = await api.post('/prescriptions', prescription);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  getPrescriptionById: async (id) => {
    try {
      const response = await api.get(`/prescriptions/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  updatePrescription: async (id, prescription) => {
    try {
      const response = await api.put(`/prescriptions/${id}`, prescription);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  deletePrescription: async (id) => {
    try {
      const response = await api.delete(`/prescriptions/${id}`);
      return response.data;
    } catch (error) {
      throw error;
    }
  },
};

export default prescriptionService;

File: Hospital Management System/backend/src/main/java/com/hospital/management/service/PrescriptionService.java
package com.hospital.management.service;

import com.hospital.management.model.Prescription;

import java.util.List;
import java.util.Optional;

public interface PrescriptionService {
    
    Prescription createPrescription(Prescription prescription);
    
    List<Prescription> getAllPrescriptions();
    
    Optional<Prescription> getPrescriptionById(Long id);
    
    Prescription updatePrescription(Long id, Prescription prescriptionDetails);
    
    void deletePrescription(Long id);
}

